guake/tests/test_guake.py

```diff
@@ -1,2 +1,106 @@
-def test_000():
-    pass
+# -*- coding: utf-8 -*-
+# pylint: disable=redefined-outer-name
+
+import json
+import os
+import pathlib
+import tempfile
+
+import pytest
+
+import guake.guake_app
+from guake.guake_app import Guake
+
+
+@pytest.fixture
+def g(mocker):
+    mocker.patch(
+        'guake.guake_app.Guake.get_xdg_config_directory',
+        return_value=pathlib.Path(tempfile.gettempdir())
+    )
+    mocker.patch('guake.guake_app.shutil.copy', create=True)
+    mocker.patch('guake.guake_app.notifier.showMessage', create=True)
+    mocker.patch('guake.guake_app.traceback.print_exc', create=True)
+    return Guake()
+
+
+def test_guake_restore_tabs(g):
+    d1 = tempfile.TemporaryDirectory()
+    d2 = tempfile.TemporaryDirectory()
+    d3 = tempfile.TemporaryDirectory()
+    d4 = tempfile.TemporaryDirectory()
+    session = {
+        "schema_version": 1,
+        "timestamp": 1556092197,
+        "workspace": {
+            "0": [
+                [
+                    {
+                        "directory": d1.name,
+                        "label": "1",
+                        "custom_label_set": True
+                    },
+                    {
+                        "directory": d2.name,
+                        "label": "2",
+                        "custom_label_set": True
+                    },
+                    {
+                        "directory": d3.name,
+                        "label": "~/dev/guake",
+                        "custom_label_set": False
+                    }
+                ]
+            ],
+            "1": [
+                [
+                    {
+                        "directory": d4.name,
+                        "label": "4",
+                        "custom_label_set": True
+                    }
+                ]
+            ]
+        }
+    }
+
+    with tempfile.NamedTemporaryFile(mode='w') as f:
+        f.write(json.dumps(session))
+        f.seek(0)
+        g.restore_tabs(os.path.basename(f.name))
+
+    nb = g.get_notebook()
+    assert nb.get_n_pages() == 3
+    assert nb.get_tab_text_index(0) == '1'
+    assert nb.get_tab_text_index(1) == '2'
+
+    nb = g.notebook_manager.get_notebook(1)
+    assert nb.get_n_pages() == 1
+    assert nb.get_tab_text_index(0) == '4'
+
+    d1.cleanup()
```

Comment:
I would try catch this to be sure to alway call cleanup even if the test breaks. Again, with fakefs, no need :)